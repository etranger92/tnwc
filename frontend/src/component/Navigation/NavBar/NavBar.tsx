import React from "react";
import "../navigation.css";
import { NavLink } from "react-router-dom";
import IconHome from "../../ElementsHTML/Icons/IconHome";
import { HashLink } from "react-router-hash-link";

type Props = {
  isItClicked: boolean;
  scroll?: boolean;
  // eslint-disable-next-line @typescript-eslint/ban-types
  clickEvent?: Function;
};
type TStyle = {
  opacity: number;
};
const NavBar: React.FC<Props> = (props) => {
  // eslint-disable-next-line react/prop-types
  const { isItClicked, scroll, clickEvent } = props;
  //When the nav scrole up we want to add a slight effect of disparition
  const style: TStyle = {
    opacity: 0,
  };
  return (
    <ul style={!isItClicked ? style : {}} id="nav-bar">
      <li>
        {" "}
        <HashLink
          to={"/#home"}
          onClick={() => {
            if (clickEvent) {
              clickEvent();
            }
            return;
          }}
        >
          {" "}
          Home
        </HashLink>
      </li>
      <li>
        {" "}
        <HashLink
          to={"/#display-article-course"}
          onClick={() => {
            if (clickEvent) {
              clickEvent();
            }
            return;
          }}
        >
          {" "}
          Course
        </HashLink>
      </li>
      <li>
        {" "}
        <NavLink
          to={"https://www.thenorthwestcollege.com/"}
          onClick={() => {
            if (clickEvent) {
              clickEvent();
            }
            return;
          }}
        >
          <IconHome effect={scroll} />
        </NavLink>
      </li>
      <li>
        {" "}
        <HashLink
          to={"/#inscription"}
          onClick={() => {
            if (clickEvent) {
              clickEvent();
            }
            return;
          }}
        >
          {" "}
          Inscription
        </HashLink>
      </li>
      <li>
        {" "}
        <HashLink
          to={"/#footer"}
          onClick={() => {
            if (clickEvent) {
              clickEvent();
            }
            return;
          }}
        >
          {" "}
          Contact
        </HashLink>
      </li>
    </ul>
  );
};

export default NavBar;
