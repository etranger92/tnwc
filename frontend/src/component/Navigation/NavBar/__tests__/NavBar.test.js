import React from "react";
import {mount} from "enzyme";
import NavBar from "../NavBar.tsx"
import { BrowserRouter as Router } from 'react-router-dom';


describe("We test the innerHTML of NavBar", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Router>
        <NavBar />
    </Router>);
  });
  test("Snapshot navBar", () => {
    expect(wrapper).toMatchSnapshot();
  });
});