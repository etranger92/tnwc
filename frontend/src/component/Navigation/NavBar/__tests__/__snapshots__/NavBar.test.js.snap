// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We test the innerHTML of NavBar Snapshot navBar 1`] = `ReactWrapper {}`;
